spec/requests/start_controller_spec.rb

```diff
@@ -16,29 +15,350 @@
   describe "GET" do
     context "when the user is not signed in" do
       it "routes user to the start page" do
-        get "/", headers: headers, params: {}
+        get root_path
         expect(path).to eq("/")
         expect(page).to have_content("Start now")
       end
     end
 
     context "when the user is signed in" do
       before do
+        allow(user).to receive(:need_two_factor_authentication?).and_return(false)
         sign_in user
       end
 
       it "routes user to the home page" do
-        get "/", headers:, params: {}
+        get root_path
         expect(page).to have_content("Welcome back")
       end
 
+      context "when testing the data displayed at the top of the home page" do
```

Comment:
I think there's something off about contexts that read "when testing" - it's a sensible grouping, I just think the description is almost trying to be another describe block rather than a context.

I can't really put my finger on it, and I'm not sure it's a rails thing to nest describe blocks in the same way I would in typescript, so I'm not that fussed - just something to think about in our general test structure I think.

Reply:
I agree with this. Looking it up, it looks liek describe and context are actually completely alias.
I was finding it awkward to be writing 'when' or 'and' at the start of these blocks
I will stop doing that and chagne here